### PR TITLE
[WIP] Make the docs assume helm3

### PR DESCRIPTION
**Pull request by GeorgianaElena**
To do list (based on @consideRatio's summary [here](https://github.com/consideRatio/consideratio.github.io/issues/6) :sparkles: ):
- [x] no need to `--purge`
- [x] new `--timeout` spec
- [x] `--cleaup-on-fail` always when upgrading
- [x] no `tiller`
- [x] make it clear that the docs assume helm3
- [x] instructions on how to migrate from helm2 to helm3 for an existing cluster
- [x] remove section about helm2 setup (or keep it for compatibility)

### CODE REVIEW

**Comment by mriedem** [comment]
FWIW we recently upgraded our testing environment hub from helm2 to helm3 and one of the things we needed to do was set `c.JupyterHub.hub_connect_ip = 'hub'` so that notebooks hit the hub API URL by hostname rather than service IP (which changed with our upgrade to helm3).

**Review by manics — COMMENTED** [review]
I've made some suggestions after a first pass over the docs.

File: doc/source/setup-jupyterhub/setup-helm.rst [hunk 1]
```diff
@@ -1,137 +1,67 @@
...
-   Client: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
-   Server: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
+   NAME    NAMESPACE       REVISION        UPDATED STATUS  CHART   APP VERSION
 
 
 Next Step
 ---------
 
-Congratulations, Helm is now set up! Let's continue with :ref:`setup-jupyterhub`!
+Helm 3 will not automatically create a namespace for JupyterHub.
```
[the remark this answers is not in the document]
Comment (consideRatio) [reply doc/source/setup-jupyterhub/setup-helm.rst:60]: I think that's a good call!

**Comment by consideRatio** [comment]
@GeorgianaElena thank you for working on this and thanks @manics for the review! :heart:

For instructors on how to migrate from Helm2 -> Helm3, I think we should save ourselves some work by referencing [the official docs](https://helm.sh/docs/topics/v2_v3_migration/) straight up even though I suggested otherwise before. They are complicated to follow, but I don't think we should spend time trying to simplify them.

**Comment by GeorgianaElena** [comment]
Thanks for reviewing this @manics and @consideRatio :heart:

**Comment by consideRatio** [comment]
Wieee! Thanks for your work on this @GeorgianaElena!

I'll go ahead and merge even though there is an error in CI because i know it is transient and fixed in current master.

**Pull request merged**